plutus-tx-plugin/test/Plugin/Data/recursive/sameEmptyRoseEval.plc.golden

```diff
@@ -7,132 +7,7 @@
-                                    [
-                                      case_EmptyRose_103
-                                      (iwrap
-                                        (lam List_118 (fun (type) (type)) (lam a_119 (type) (all out_List_120 (type) (fun out_List_120 (fun (fun a_119 (fun [List_118 a_119] out_List_120)) out_List_120)))))
-                                        a_117
-                                        (abs
-                                          out_List_121
-                                          (type)
-                                          (lam
-                                            case_Nil_122
-                                            out_List_121
-                                            (lam
-                                              case_Cons_123
-                                              (fun a_117 (fun [(lam a_124 (type) (ifix (lam List_125 (fun (type) (type)) (lam a_126 (type) (all out_List_127 (type) (fun out_List_127 (fun (fun a_126 (fun [List_125 a_126] out_List_127)) out_List_127))))) a_124)) a_117] out_List_121))
-                                              case_Nil_122
-                                            )
-                                          )
-                                        )
-                                      )
-                                    ]
-                                  )
-                                )
-                              )
-                            ]
-                            (iwrap
-                              (lam List_118 (fun (type) (type)) (lam a_119 (type) (all out_List_120 (type) (fun out_List_120 (fun (fun a_119 (fun [List_118 a_119] out_List_120)) out_List_120)))))
-                              a_117
-                              (abs
-                                out_List_121
-                                (type)
-                                (lam
-                                  case_Nil_122
-                                  out_List_121
-                                  (lam
-                                    case_Cons_123
-                                    (fun a_117 (fun [(lam a_124 (type) (ifix (lam List_125 (fun (type) (type)) (lam a_126 (type) (all out_List_127 (type) (fun out_List_127 (fun (fun a_126 (fun [List_125 a_126] out_List_127)) out_List_127))))) a_124)) a_117] out_List_121))
-                                    case_Nil_122
-                                  )
-                                )
-                              )
-                            )
-                          ]
-                        )
-                      )
-                    )
-                  )
-                ]
-              )
-            )
-          )
-        )
-      ]
+      [ case_EmptyRose_103 arg_0_93 ]
```

Comment:
I feel like I've broken something here. Will need to check.

Reply:
Yep, I did: I replaced the CK machine with the CEK machine and the latter doesn't discharge the environment, so we get uninsantiated variables in the output. @michaelpj

Reply:
Hm, interesting. I think the CEK machine probably should discharge the environment at the end. It doesn't matter if we're a) only testing for the presence of error or b) only computing to constants, but it would be nice if this kind of generic reduction worked.

Perhaps walk back the change here and implement that separately?

Reply:
> Perhaps walk back the change here

Done.